nixos/modules/system/boot/loader/generic-extlinux-compatible/extlinux-conf-builder.sh

```diff
@@ -96,7 +97,12 @@ addEntry() {
     echo "  LINUX ../nixos/$(basename $kernel)"
     echo "  INITRD ../nixos/$(basename $initrd)"
     if [ -d "$dtbDir" ]; then
-        echo "  FDTDIR ../nixos/$(basename $dtbs)"
+        # if a dtbName was specified explicitly, use that, else use FDTDIR
+        if [ -n "$dtbName" ]; then
+            echo "  FDT ../nixos/$(basename $dtbs)/${dtbName}.dtb"
+        else
+            echo "  FDTDIR ../nixos/$(basename $dtbs)"
+        fi
```

Comment:
This is a bit better than the previous, but has the fatal flaw of not failing if `-n` is given, but `$dtbDir` isn't a thing.

I guess in a new `else` branch, fail on `[ -n "$dtbName" ]` wasn't given.

Reply:
well, it also doesn't check for the fdt to be really present, and even with a specific fdt specified, still copies all of them.

Not sure how much logic we actually want to add in here…

Reply:
I added the new else branch as described.